vllm/entrypoints/llm.py

```diff
@@ -1045,37 +1037,92 @@ def ensure_str(prompt: SingletonPrompt):
         if len(text_1) == 1:
```

Comment:
We only want do this for cross encoder models. For embedding models where each prompt is sent on it's own, we don't want to send the prompt N times.